must-call-checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,144 @@
+package org.checkerframework.checker.mustcall;
+
+import static org.checkerframework.common.value.ValueCheckerUtils.getValueOfAnnotationWithStringArgument;
+
+import java.lang.annotation.Annotation;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.util.Elements;
+import org.checkerframework.checker.mustcall.qual.MustCall;
+import org.checkerframework.checker.mustcall.qual.MustCallTop;
+import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.framework.type.ElementQualifierHierarchy;
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.AnnotationBuilder;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+/**
+ * The annotated type factory for the must call checker. Primarily responsible for the subtyping
+ * rules between @MustCall annotations.
+ */
+public class MustCallAnnotatedTypeFactory extends BaseAnnotatedTypeFactory {
+
+  /** The top annotation. */
+  final AnnotationMirror TOP;
+
+  /** The bottom annotation. */
+  final AnnotationMirror BOTTOM;
+
+  /**
+   * Default constructor matching super. Should be called automatically.
+   *
+   * @param checker the checker associated with this type factory
+   */
+  public MustCallAnnotatedTypeFactory(final BaseTypeChecker checker) {
+    super(checker);
+    TOP = AnnotationBuilder.fromClass(elements, MustCallTop.class);
+    BOTTOM = createMustCall();
```

Comment:
Why don't we need `@MustCallBottom`?  We had `@CalledMethodsBottom`, distinguished from `@CalledMethods({})`.

Reply:
`@CalledMethodsTop` is equivalent to `@CalledMethods({})`. We have `@CalledMethodsBottom` for the same reason that `@MustCallTop` is required here: there are a potentially-infinite number of `@MustCall(...)` annotations, and they need to have a single supertype; similarly, there are a potentially-infinite number of `@CalledMethods(...)` types, and they need a single subtype. The difference is the direction of the type hierarchy: `@MustCall({})` is the bottom type, but `@CalledMethods({})` is the top type.

We actually don't need `@CalledMethodsTop`: it was included so that predicates had a sensible LUB with `@CalledMethods` annotations, but even that isn't needed. The Called Methods Checker in the CF master branch doesn't have an `@CalledMethodsTop` annotation and instead just uses `@CalledMethods({})` as the top type.

Reply:
That makes sense, thanks!